src/PeerConnection.cpp

```diff
@@ -92,7 +92,9 @@ namespace mediasoupclient
 
 			this->peerConnectionFactory = webrtc::CreatePeerConnectionFactory(
 			  this->networkThread.get(),
-			  this->workerThread.get(),
+			  // TMP: Use the same  thread for signaling and worker.
+			  // this->workerThread.get(),
+			  this->signalingThread.get(),
```

Comment:
I also want to note that this looks extremely sketchy. If LibWebRTC truly wanted you to use signaling thread as both worker and signaling, they'd do so in other ways (such as taking only 2 pointers).

Reply:
Yep, this is a temporal work around that needs to be removed.